python/test/unit/la/test_casters.py

```diff
@@ -0,0 +1,75 @@
+"""Unit tests for PETSc casters"""
+
+# Copyright (C) 2019 Francesco Ballarin
+#
+# This file is part of DOLFIN (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+
+import dolfin
+import numpy
+import petsc4py
+import os
+import pytest
+import sys
+from petsc4py import PETSc
+from dolfin.jit import dolfin_pc, mpi_jit_decorator
+from dolfin_utils.test.fixtures import tempdir  # noqa: F401
+
+
+def test_petsc_casters_cppimport(tempdir):  # noqa: F811
+    """
+    Test casters of PETSc objects in codes compiled with cppimport
+    """
+
+    cppimport = pytest.importorskip("cppimport")
+
+    @mpi_jit_decorator
+    def compile_module():
+        cpp_code_header = f"""
+        <%
+        setup_pybind11(cfg)
+        cfg['include_dirs'] += {dolfin_pc["include_dirs"] + [petsc4py.get_include()]}
+        cfg['compiler_args'] += {["-D" + dm for dm in dolfin_pc["define_macros"]]}
+        cfg['libraries'] += {dolfin_pc["libraries"]}
```

Comment:
This is probably overkill as well, as only the petsc libraries are needed, but not dolfin itself or the other dependencies. However, rather than fiddling with PETSc's pc, I'd leave it as is for future users that may need to use DOLFIN classes as well.